### PR TITLE
[web] init history if null

### PR DESCRIPTION
**Pull request by avatus**
TL;DR: this will make sure that history is init before using any of it's methods. 

This fix is only to fix our broken Storybook tests. The old resources have a check to push to the Unified Resources route. this uses our `history` utility. We've never used this in Storybook tests and the old resource stories currently break because it's trying to call a method on a null instance.

I added it to the methods that called `_inst` but tbh it was only needed in `replace` for this fix. but might as well fix the others in case it come up later.

"Why is it pushing to the unified resources route?"
Well... it is and it isnt. the route doesn't matter in the storybook tests. So, this is "hacky" but I wasn't able to find a reliable way to
1. set and, more importantly, remove a localStorage variable
2. and do that PER TEST. there are decorators yes, but those don't have any "cleanup" functions. 

The storybook _tests_ themselves don't have this problem as i [set a beforeAll and afterAll](https://github.com/gravitational/teleport/commit/d9f7394194aee5b97c020fe53c25cb492c3234a0) to skip this history check, but for rendering storybook tests themselves, i haven't found a way to do this.

Happy to dive in deeper, but this seems like a simple fix that gets what we want done which is, everything still pushing around the way we want, and storybooks aren't broken anymore

### CODE REVIEW

**Review by bl-nero — APPROVED** [review]

File: web/packages/teleport/src/services/history/history.ts
Comment (ravicious) [inline web/packages/teleport/src/services/history/history.ts:1]: I feel like this is not the best way to go about it. If something is needed only for a couple of stories, I'd usually recommend wrapping them in some kind of component that runs the necessary code.

See [my comment](https://github.com/gravitational/teleport/pull/27788#discussion_r1231063745) and [how it was addressed](https://github.com/gravitational/teleport/pull/27788/commits/e019fc15872c3891e2f16c459016ed501ddc198f). Originally, Grzegorz mocked the whole `AppContext` which potentially had cascading effects on all stories. We only needed to set the theme, but now we're potentially affecting the behavior of other stories. We ended up inserting a static theme provider instead of mocking the whole AppContext.

But that's not something we can do here. I'm looking at how `history` is used and it's a global object. In the real app, `history.init()` is called in `boot.tsx`. Wouldn't it be better to follow suit and initialize `history` before any stories are run?

```diff
diff --git a/web/.storybook/preview.js b/web/.storybook/preview.js
index 874d135586..99902b747d 100644
--- a/web/.storybook/preview.js
+++ b/web/.storybook/preview.js
@@ -31,6 +31,7 @@ import {
   lightTheme as teletermLightTheme,
 } from './../packages/teleterm/src/ui/ThemeProvider/theme';
 import { handlersTeleport } from './../packages/teleport/src/mocks/handlers';
+import history from './../packages/teleport/src/services/history/history';
 import { UserContextProvider } from 'teleport/User';
 
 // Checks we are running non-node environment (browser)
@@ -42,6 +43,8 @@ if (typeof global.process === 'undefined') {
   window.msw = { worker, rest };
 }
 
+history.init();
+
 // wrap each story with theme provider
 const ThemeDecorator = (storyFn, meta) => {
   let ThemeProvider;

```

Then you don't need to modify the implementation just to fix stories.

Comment (avatus) [reply web/packages/teleport/src/services/history/history.ts:1]: Sometimes all it takes is another set of eyes 👁️ 👄 👁️ 

Thank you, this is exactly what i was trying to do with the decorators but ended up missing _this_ way, and tried every other way under the sun. Much better

**Review by ravicious — APPROVED** [review]
